buildpacks/nodejs-npm-install/src/main.rs

```diff
@@ -0,0 +1,217 @@
+mod errors;
+mod layers;
+mod npm;
+
+use crate::errors::NpmInstallBuildpackError;
+use crate::layers::npm_cache::NpmCacheLayer;
+use commons::fun_run::CommandWithName;
+use commons::output::build_log::{BuildLog, Logger, SectionLogger};
+use commons::output::fmt;
+use commons::output::section_log::{log_step, log_step_stream};
+use commons::output::warn_later::WarnGuard;
+use heroku_nodejs_utils::application;
+use heroku_nodejs_utils::package_json::PackageJson;
+use heroku_nodejs_utils::package_manager::PackageManager;
+use heroku_nodejs_utils::vrs::Version;
+use libcnb::build::{BuildContext, BuildResult, BuildResultBuilder};
+use libcnb::data::build_plan::BuildPlanBuilder;
+use libcnb::data::launch::{LaunchBuilder, ProcessBuilder};
+use libcnb::data::{layer_name, process_type};
+use libcnb::detect::{DetectContext, DetectResult, DetectResultBuilder};
+use libcnb::generic::{GenericMetadata, GenericPlatform};
+use libcnb::{buildpack_main, Buildpack, Env};
+use std::io::{stdout, Stdout};
+use std::path::Path;
+use std::process::Command;
+
+pub(crate) const BUILDPACK_NAME: &str = "Heroku Node.js npm Install Buildpack";
+
+pub(crate) struct NpmInstallBuildpack;
+
+impl Buildpack for NpmInstallBuildpack {
+    type Platform = GenericPlatform;
+    type Metadata = GenericMetadata;
+    type Error = NpmInstallBuildpackError;
+
+    fn detect(&self, context: DetectContext<Self>) -> libcnb::Result<DetectResult, Self::Error> {
+        let package_json_exists = context
+            .app_dir
+            .join(PackageManager::Npm.lockfile())
```

Comment:
The variable name is `package_json_exists`, however the file being checked instead seems to be the lockfile? (`PackageManager::Npm.lockfile()`). I presume the latter is correct, so the variable name is what needs changing?